components/basecomponents/Navigation/index.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { HeaderNavigation, HEADER_NAVIGATION } from 'data/headerNavigation';
+import { useRouter } from 'next/router';
+import { useClickOutside } from 'components/hooks/useClickOutside';
+import { useTranslation } from 'next-i18next';
+import clsx from 'clsx';
+import Link from 'next/link';
+import { getCountryVariant } from 'utils/locales';
+import { Dropdown } from './Dropdown';
+
+export interface OpenDropdown {
+  open: boolean;
+  id?: string;
+}
+
+interface NavigationItems extends HeaderNavigation {
+  active?: boolean;
+}
+
+export const Navigation: React.FC<{ showNavigation: boolean }> = ({ showNavigation }) => {
+  const { asPath } = useRouter();
+  const { t } = useTranslation('common');
+  const [dropdown, setDropdown] = React.useState<OpenDropdown>({ open: false });
+  const { ref } = useClickOutside<HTMLDivElement>(() => setDropdown({ open: false }));
+
+  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    const id = e.currentTarget.id;
+
+    setDropdown({ open: !dropdown.open, id });
+  };
+
+  const items: NavigationItems[] = HEADER_NAVIGATION.map((item) => {
+    const { link } = item;
+
+    if (new RegExp(link).test(asPath)) {
+      return {
+        ...item,
+        active: true,
+      };
+    }
+
+    return { ...item };
+  });
+
+  return (
+    <nav className="w-full pr-10">
+      <div className="container flex flex-wrap justify-end items-center mx-auto" ref={ref}>
```

Comment:
1. ref should be named more specifically `clickOutsideRef` or something
2. it should not be attached here to the menu container, but to the dropdown container. Right now you have a bug where clicking on a menu item does not close the dropdown. 
3. The click outside logic should be handled inside the Dropdown component entirely.

Reply:
Ref should be named more specifically if you have more than one ref in component.